src/services/summary/hooks/trigger-calculation.js

```diff
@@ -0,0 +1,88 @@
+const _ = require('lodash');
+const moment = require('moment');
+const ActivityLog = require('../../activity-log/activity-log-model');
+const actionConstants = require('../../../constants/actions');
+const Summary = require('../../summary/summary-model');
+
+// triggerCalculation is a before hook function to trigger a summary
+// calculation on specified dates and organization.
+// If no date specifies, it will calculate today's summary.
+const triggerCalculation = () => (hook) => {
+  const startDate = hook.params.query.start_date;
+  const endDate = hook.params.query.end_date;
+  const organizationName = hook.params.query.organization;
+  const trigger = hook.params.query.trigger;
+  if (!trigger) {
+    return Promise.resolve(hook);
+  }
+  let timestamp = {};
+  // Change start_date and end_date to mongoose query format.
+  if (startDate) {
+    timestamp.$gte = startDate;
+  }
+  if (endDate) {
+    timestamp.$lte = endDate;
+  }
+  if (_.isEmpty(timestamp)) {
+    timestamp = moment().format('YYYY-MM-DD');
+  }
+  // Start a calculation
+  const query = {
+    organization: organizationName,
+    timestamp,
+  };
+  // Remap state to summary's field name.
+  const remapStateToFieldName = {};
+  remapStateToFieldName[actionConstants.ASSIGN] = 'assigned';
+  remapStateToFieldName[actionConstants.PROCESS] = 'processing';
+  remapStateToFieldName[actionConstants.RESOLVE] = 'resolved';
```

Comment:
yes, it is a mapping to calculate the number of happening activity in the end from activity_log.